tests/test_requests.py

```diff
@@ -2585,5 +2585,11 @@ def test_post_json_nan(self, httpbin):
 
     def test_json_decode_compatibility(self, httpbin):
         r = requests.get(httpbin('bytes/20'))
-        with pytest.raises(requests.exceptions.JSONDecodeError):
+        data = r.text
+        with pytest.raises(requests.exceptions.JSONDecodeError) as excinfo:
             r.json()
+        assert isinstance(excinfo.value, RequestException)
+        assert isinstance(excinfo.value, JSONDecodeError)
+        assert data not in str(excinfo.value)
+        if is_py3:
```

Comment:
Also somewhat of a nit. It may be better to just make this its own test with a `@unittest.skipIf(not is_py3)` at the top.

e.g.

```python
@unittest.skipIf(not is_py3)
def test_json_decode_persists_doc_attr(self, httpbin):
    r = requests.get(httpbin('bytes/20'))
    with pytest.raises(requests.exceptions.JSONDecodeError) as e:
        r.json()
    assert e.doc == r.text
```

Reply:
don't we want to test that the py2 object is also what we expect? Or if we don't can i can make it py3 only?

Reply:
The original test doesn't assert anything about Python 2 either. So we don't lose anything with this change. I suppose we could assert doc _isn't_ on the Python 2 exception but I'm not sure how meaningful that is.

Reply:
ok, i think i updated it with what you had in mind.